library/src/interactor_impl.cxx

```diff
@@ -268,6 +282,88 @@ class interactor_impl::internals
     self->Window.render();
   }
 
+  static void OnMiddleButtonPress(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
+    self->MiddleClickPosision[0] = click_pos[0];
+    self->MiddleClickPosision[1] = click_pos[1];
+
+    self->Style->OnMiddleButtonDown();
+  }
+
+  static void OnMiddleButtonRelease(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
+
+    const auto x_delta = click_pos[0] - self->MiddleClickPosision[0];
+    const auto y_delta = click_pos[1] - self->MiddleClickPosision[1];
+    const auto sq_pos_delta = x_delta * x_delta + y_delta * y_delta;
+    if (sq_pos_delta < self->DragDistanceTol * self->DragDistanceTol)
+    {
+      bool pick_successful = false;
+      double picked_pos[3];
+      self->CellPicker->Pick(self->MiddleClickPosision[0], self->MiddleClickPosision[1], 0,
+        self->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer());
+      if (self->CellPicker->GetActors()->GetNumberOfItems() > 0)
+      {
+        self->CellPicker->GetPickPosition(picked_pos);
+        pick_successful = true;
+      }
+      else
+      {
+        self->PointPicker->Pick(self->MiddleClickPosision[0], self->MiddleClickPosision[1], 0,
+          self->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer());
+        if (self->PointPicker->GetActors()->GetNumberOfItems() > 0)
+        {
+          self->PointPicker->GetPickPosition(picked_pos);
+          pick_successful = true;
+        }
+      }
+
+      if (pick_successful)
+      {
+        const auto pos = self->Window.getCamera().getPosition();
+        const auto foc = self->Window.getCamera().getFocalPoint();
+
+        const auto dx = picked_pos[0] - foc[0];
+        const auto dy = picked_pos[1] - foc[1];
+        const auto dz = picked_pos[2] - foc[2];
+
+        if (self->TransitionDuration > 0)
+        {
+          const auto start = std::chrono::high_resolution_clock::now();
+          const auto end = start + std::chrono::milliseconds(self->TransitionDuration);
+          auto now = start;
+          while (now < end)
+          {
+            const double linear_t =
```

Comment:
We are using camel case for variables. What about `linearTime` instead of `linear_t`.

Reply:
This is the hardest change of the bunch. They're interpolation parameters, can we just name then `t` and `u`?

Reply:
sure